docs/gallery/plot_aligneddynamictable.py

```diff
@@ -0,0 +1,323 @@
+_ = customer_table.get_category(name='home_address')
+
+# Alternatively, we can use normal array slicing to get the category as a pandas DataFrame.
+# NOTE: In contrast to the previous call, the table is here converted to a DataFrame.
+_ = customer_table['home_address']
+
+###############################################################################
+# Accessing columns
+# ^^^^^^^^^^^^^^^^^
+# We can use the standard Python ``in`` operator to check if a column exists
+
+# To check if a column exists in the primary table we only need to specify the column name
+# or alternatively specify the category as None
+_ = 'firstname' in customer_table
+_ = (None, 'firstname') in customer_table
+# To check if a column exists in a category table we need to specify the category
+# and column name as a tuple
+_ = ('home_address', 'zipcode') in customer_table
+
+
+###############################################################################
+# We can use standard array slicing to get the :py:class:`~hdmf.common.table.VectorData` object of a column.
+
+# To get a column from the primary table we just provide the name.
+_ = customer_table['firstname']
+# To get a column from a category table we provide both the category name and column name
+_ = customer_table['home_address', 'city']
+
+
+###############################################################################
+# Accessing rows
+# ^^^^^^^^^^^^^^
+#
+# Accessing rows works much like in :doc:`DynamicTable <dynamictable>`
+#
+
+# Get a single row by index as a DataFrame
+customer_table[1]
+
+###############################################################################
+#
+
+# Get a range of rows as a DataFrame
+customer_table[0:2]
+
+###############################################################################
+#
+
+# Get a list of rows as a DataFrame
+customer_table[[0, 2]]
+
+
+###############################################################################
+# Accessing cells
+# ^^^^^^^^^^^^^^^
+#
+# To get a set of cells we need to specify the: 1) category, 2) column, and 3) row index when slicing into the table.
+#
+# When selecting from the primary table we need to specify None for the category, followed by the column name and
+# the selection.
+
+# Select rows 0:2 from the 'firstname' column in the primary table
+customer_table[None, 'firstname', 0:2]
+
+###############################################################################
+#
+
+# Select rows 1 from the 'firstname' column in the primary table
+customer_table[None, 'firstname', 1]
+
+###############################################################################
+#
+
+# Select rows 0 and 2 from the 'firstname' column in the primary table
+customer_table[None, 'firstname', [0, 2]]
+
+###############################################################################
+#
+
+# Select rows 0:2 from the 'city' column of the 'home_address' category table
+customer_table['home_address', 'city', 0:2]
```

Comment:
Is it possible to have nested AlignedDynamicTables and if so, what is the syntax for accessing a single row of that table? Would it be:
```python
customer_table[category, sub-category, column, indices]
```
?

Reply:
That's a great question. I'm honestly not sure, I've never tried. Since ``AlignedDynamicTable`` is a ``DynamicTable`` the schema would in principle allow nesting AlignedDynamicTables but I'm not sure this would work right now in the code. I'll give it a try to see if I can make this work and if not add checks to raise errors when creating categories to raise an error if this is not allowed.

Reply:
Creating such a table where ``AlignedDynamicTable`` is a category of an ``AlignedDynamicTable`` in principle works, but dealing with this in ``AlignedDynamicTable.__getitem__``, ``AlignedDynamicTable.to_dataframe`` etc. does not. Supporting this should be possible but it's not a trivial fix. I'll add error checks for now to prevent this from happening and we can address this in a separate issue/PR.

Reply:
I have created issue https://github.com/hdmf-dev/hdmf/issues/574 to describe this problem and added the following fixes here: 1) raise ValueError in __init__ and add_category if an AlignedDynamicTable is given, 2) updated the docstring and tutorial to mention this behavior, 3) add test cases to make sure the errors are being raised as expected. I also created (and closed) https://github.com/hdmf-dev/hdmf/issues/575 for future documentation purposed.